packages/suite/src/hooks/wallet/form/useUtxoSelection.ts

```diff
@@ -76,23 +87,76 @@ export const useUtxoSelection = ({
         composeRequest,
     ]);
 
+    const sortUtxos = (utxos: AccountUtxo[]): AccountUtxo[] => {
+        if (!utxoSorting) {
+            return utxos;
+        }
+
+        const sortFromNewestToOldest = (a: AccountUtxo, b: AccountUtxo) => {
+            let valueA: number;
+            let valueB: number;
+
+            if (a.blockHeight > 0 && b.blockHeight > 0) {
+                valueA = a.blockHeight;
+                valueB = b.blockHeight;
+            } else {
+                // Pending transactions do not have blockHeight, so we must use blockTime of the transaction instead.
+                const getBlockTime = (txid: string) => {
+                    const transaction = accountTransactions.find(
+                        transaction => transaction.txid === txid,
+                    );
+
+                    return transaction?.blockTime ?? 0;
+                };
+                valueA = getBlockTime(a.txid);
+                valueB = getBlockTime(b.txid);
+            }
+
+            return new BigNumber(valueB ?? 0).comparedTo(new BigNumber(valueA ?? 0));
+        };
+
+        const sortFromLargestToSmallest = (a: AccountUtxo, b: AccountUtxo) =>
+            new BigNumber(b.amount).comparedTo(new BigNumber(a.amount));
+
+        // Manipulating the array directly would cause a runtime error, so we create a copy.
+        const utxosForSorting = utxos.slice();
+
+        switch (utxoSorting) {
+            case 'newestFirst':
+                return utxosForSorting.sort(sortFromNewestToOldest);
+            case 'oldestFirst':
+                return utxosForSorting.sort(sortFromNewestToOldest).reverse();
```

Comment:
I used `reverse` because in case when the compared values are the same, I want the items to reverse order when I switch e.g. from newest first to oldest first. I find that to be the expected behaviour. See the two UTXOs from Oct 11 on your branch:
![Screenshot 2024-12-03 at 14 38 24](https://github.com/user-attachments/assets/ce460e83-ba63-4e74-8d42-328f39cc9327)
![Screenshot 2024-12-03 at 14 38 18](https://github.com/user-attachments/assets/1c3adde1-18fc-440e-8cbf-b5795a996992)